### ISSUE TITLE
accuracy output wrong when having multiple binary outputs

### ISSUE THREAD

**Issue by BaldwinTheThird** [fchollet/keras#2522]
There are use cases where it makes sense to have multiple binary outputs. Models like this can be currently set up in Keras using output_dim = dimension of data and choosing 'binary_crossentropy' as loss function. When running a model like this the following lines of code cause problems (models.py starting at line 525):

```
        # set class_mode, for accuracy computation:
        if self.output_shape[-1] == 1:
            class_mode = 'binary'
        else:
            class_mode = 'categorical'
        self.class_mode = class_mode

        if class_mode == 'categorical':
            train_accuracy = K.mean(K.equal(K.argmax(self.y, axis=-1),
                                            K.argmax(self.y_train, axis=-1)))
            test_accuracy = K.mean(K.equal(K.argmax(self.y, axis=-1),
                                           K.argmax(self.y_test, axis=-1)))
        elif class_mode == 'binary':
            if self.loss.__name__ == 'categorical_crossentropy':
                warnings.warn('Your model output has shape ' + str(self.output_shape) +
                              ' (1-dimensional features), but you are using ' +
                              ' the `categorical_crossentropy` loss. You ' +
                              'almost certainly want to use `binary_crossentropy` instead.')
            train_accuracy = K.mean(K.equal(self.y, K.round(self.y_train)))
            test_accuracy = K.mean(K.equal(self.y, K.round(self.y_test)))
```

With multiple outputs, output_shape [-1] would not be equal to 1 and 'binary' would have to apply. An easy hack is to just always set class_mode = 'binary' when dealing with a model with multiple binary outputs. Obviously, it will then break when there really is categorical data.

A related issue is that sometimes it would even make sense to have multiple categorical outputs (multiple softmax). It doesn't seem like Keras has been designed with this possibility in mind. It would be useful to be able to feed as input a 2-D array of integers in the range [1,number_categories] where each column of the array represents an output. Is that something that you are thinking about adding?

**Comment by braingineer** [comment]
>  It doesn't seem like Keras has been designed with this possibility in mind.

I think I'm not understanding something. 

If you have:

```python
predictions1 = Dense(some_num, activation='softmax')(everything_before)
predictions2 = Dense(some_num, activation='softmax')(everything_before)
model = Model(inputs=whatever, outputs=[predictions1, predictions2])
model.compile(losses=['categorical_crossentropy', 'categorical_crossentropy'])
```

Doesn't this do two categorical outputs, two softmaxes, and expect two sets of y_pred variables?
